src/Messages/OWSMessageSender.m

```diff
@@ -862,8 +865,7 @@ - (void)sendMessage:(TSOutgoingMessage *)message
                         return failureHandler(error, YES);
                     }
 
-                    [self handleMismatchedDevices:serializedResponse recipient:recipient];
-                    retrySend();
+                    [self handleMismatchedDevices:serializedResponse recipient:recipient completion:retrySend];
```

Comment:
Correct me if I'm wrong but won't these changes move invocation of `retrySend()` to the `sessionStoreQueue`?  Is that okay?  Should we modify `handleMismatchedDevices` and `handleStaleDevicesWithResponse` to execute their new completion handlers on another queue?